daemons/based/based_callbacks.c

```diff
@@ -1254,6 +1280,15 @@ cib_process_command(xmlNode * request, xmlNode ** reply, xmlNode ** cib_diff, gb
             cib__clear_call_options(call_options, "call", cib_zero_copy);
         }
 
+        /* Calculate the hash value of the section before the change. */
+        if (pcmk__str_eq(CIB_OP_REPLACE, op, pcmk__str_none)) {
+            current_config_digest = calculate_section_digest("//" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION, current_cib);
+            current_nodes_digest = calculate_section_digest("//" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION "/" XML_CIB_TAG_NODES, current_cib);
+            current_alerts_digest = calculate_section_digest("//" XML_TAG_CIB "/" XML_CIB_TAG_CONFIGURATION "/" XML_CIB_TAG_ALERTS, current_cib);
```

Comment:
I'm not sure whether we should track config_digest.

Internally, pacemaker (via attrd_cib_replaced_cb() and do_cib_replaced()) only cares about the nodes and alerts sections, so we could eliminate config_digest.

On the other hand, if any external code uses the replace notifications (I'm not aware of any), it might rely on the existing behavior of tracking whether the entire configuration is replaced. In that case, we could keep config_digest, and we could drop alerts_digest since any changes there would show up in the config_digest already (nodes_digest will still come in handy for the election purposes).

I lean to keeping the nodes/alerts digests and dropping the config digest, since that will result in fewer unnecessary callbacks for changes that don't matter to attrd or controld. In the future, I could see modifying the API so that the caller specifies a list of sections they want to be called for, but that's not necessary now.

Reply:
I actually thought so.
I think it's enough to notify you of changes in node and alert.

Let's abolish comparing and notifying digests of config.

Many thanks,
Hideo Yamauchi.